internal/controller/drpolicy_peerclass.go

```diff
@@ -0,0 +1,590 @@
+}
+
+func pruneSClassViews(
+	m util.ManagedClusterViewGetter,
+	log logr.Logger,
+	clusterName string,
+	survivorClassNames []string,
+) error {
+	mcvList, err := m.ListSClassMCVs(clusterName)
+	if err != nil {
+		return err
+	}
+
+	return pruneClassViews(m, log, clusterName, survivorClassNames, mcvList)
+}
+
+// getSClassesFromCluster gets StorageClasses that are claimed in the ManagedClusterInstance
+func getSClassesFromCluster(
+	u *drpolicyUpdater,
+	m util.ManagedClusterViewGetter,
+	mc *util.ManagedClusterInstance,
+	clusterName string,
+) ([]*storagev1.StorageClass, error) {
+	sClasses := []*storagev1.StorageClass{}
+
+	sClassNames := mc.StorageClassClaims()
+	if len(sClassNames) == 0 {
+		return sClasses, nil
+	}
+
+	annotations := make(map[string]string)
+	annotations[AllDRPolicyAnnotation] = clusterName
+
+	for _, scName := range sClassNames {
+		sClass, err := m.GetSClassFromManagedCluster(scName, clusterName, annotations)
+		if err != nil {
+			return []*storagev1.StorageClass{}, err
+		}
+
+		sClasses = append(sClasses, sClass)
+	}
+
+	return sClasses, pruneSClassViews(m, u.log, clusterName, sClassNames)
+}
+
+// getClusterClasses inspects, using ManagedClusterView, the ManagedCluster claims for all storage related classes,
+// and returns the set of classLists for the passed in clusters
+func getClusterClasses(
+	u *drpolicyUpdater,
+	m util.ManagedClusterViewGetter,
+	cluster string,
+) (classLists, error) {
+	mc, err := util.NewManagedClusterInstance(u.ctx, u.client, cluster)
+	if err != nil {
+		return classLists{}, err
+	}
+
+	clID, err := mc.ClusterID()
+	if err != nil {
+		return classLists{}, err
+	}
+
+	sClasses, err := getSClassesFromCluster(u, m, mc, cluster)
+	if err != nil || len(sClasses) == 0 {
+		return classLists{}, err
+	}
+
+	vsClasses, err := getVSClassesFromCluster(u, m, mc, cluster)
+	if err != nil || len(sClasses) == 0 {
+		return classLists{}, err
+	}
+
+	vrClasses, err := getVRClassesFromCluster(u, m, mc, cluster)
+	if err != nil || len(sClasses) == 0 {
+		return classLists{}, err
+	}
+
+	return classLists{
+		clusterID: clID,
+		sClasses:  sClasses,
+		vrClasses: vrClasses,
+		vsClasses: vsClasses,
+	}, nil
+}
+
+// deleteViewsForClasses deletes all views created for classes for the passed in clusterName
+func deleteViewsForClasses(m util.ManagedClusterViewGetter, log logr.Logger, clusterName string) error {
+	if err := pruneSClassViews(m, log, clusterName, []string{}); err != nil {
+		return err
+	}
+
+	if err := pruneVSClassViews(m, log, clusterName, []string{}); err != nil {
+		return err
+	}
+
+	return pruneVRClassViews(m, log, clusterName, []string{})
+}
+
+// updatePeerClasses inspects required classes from the clusters that are part of the DRPolicy and updates DRPolicy
+// status with the peer information across these clusters
+func updatePeerClasses(u *drpolicyUpdater, m util.ManagedClusterViewGetter) error {
+	cls := []classLists{}
+
+	if len(u.object.Spec.DRClusters) <= 1 {
```

Comment:
(nit) just be specific; if `drClusters != 2`. That would prevent the length of DRClusters greater than 2 as well.

Reply:
I am (currently) writing this to deal with more than two clusters. Although it is a simple latch to prevent that case happening, and can add it.

There are many parts of the code where we ensure there are only 2 clusters, so maybe I can add a comment than change the code here? (on the fence about this one)